### ISSUE TITLE
Travis CI on develop branch failed

### ISSUE THREAD

**Issue by wangkuiyi** [sql-machine-learning/sqlflow#2318]
It seems that after merging, the Travis CI on the develop branch failed https://travis-ci.com/github/sql-machine-learning/sqlflow/jobs/337411228#L1250-L1335

```
=== RUN   TestConvergenceAndAccuracy
--- FAIL: TestConvergenceAndAccuracy (23.06s)
    convergence_test.go:76: 
        	Error Trace:	convergence_test.go:76
        	            				convergence_test.go:56
        	Error:      	Not equal: 
        	            	expected: 1
        	            	actual  : 0
        	Test:       	TestConvergenceAndAccuracy
    convergence_test.go:76: 
        	Error Trace:	convergence_test.go:76
        	            				convergence_test.go:56
        	Error:      	Not equal: 
        	            	expected: 0
        	            	actual  : 1
        	Test:       	TestConvergenceAndAccuracy
    convergence_test.go:76: 
        	Error Trace:	convergence_test.go:76
        	            				convergence_test.go:56
        	Error:      	Not equal: 
        	            	expected: 1
        	            	actual  : 0
        	Test:       	TestConvergenceAndAccuracy
    convergence_test.go:76: 
        	Error Trace:	convergence_test.go:76
        	            				convergence_test.go:56
        	Error:      	Not equal: 
        	            	expected: 0
        	            	actual  : 1
        	Test:       	TestConvergenceAndAccuracy
    convergence_test.go:76: 
        	Error Trace:	convergence_test.go:76
        	            				convergence_test.go:56
        	Error:      	Not equal: 
        	            	expected: 1
        	            	actual  : 0
        	Test:       	TestConvergenceAndAccuracy
    convergence_test.go:76: 
        	Error Trace:	convergence_test.go:76
        	            				convergence_test.go:56
        	Error:      	Not equal: 
        	            	expected: 2
        	            	actual  : 1
        	Test:       	TestConvergenceAndAccuracy
    convergence_test.go:76: 
        	Error Trace:	convergence_test.go:76
        	            				convergence_test.go:56
        	Error:      	Not equal: 
        	            	expected: 0
        	            	actual  : 1
        	Test:       	TestConvergenceAndAccuracy
    convergence_test.go:76: 
        	Error Trace:	convergence_test.go:76
        	            				convergence_test.go:56
        	Error:      	Not equal: 
        	            	expected: 1
        	            	actual  : 0
        	Test:       	TestConvergenceAndAccuracy
    convergence_test.go:76: 
        	Error Trace:	convergence_test.go:76
        	            				convergence_test.go:56
        	Error:      	Not equal: 
        	            	expected: 3
        	            	actual  : 2
        	Test:       	TestConvergenceAndAccuracy
    convergence_test.go:76: 
        	Error Trace:	convergence_test.go:76
        	            				convergence_test.go:56
        	Error:      	Not equal: 
        	            	expected: 0
        	            	actual  : 1
        	Test:       	TestConvergenceAndAccuracy
    convergence_test.go:76: 
        	Error Trace:	convergence_test.go:76
        	            				convergence_test.go:56
        	Error:      	Not equal: 
        	            	expected: 1
        	            	actual  : 0
        	Test:       	TestConvergenceAndAccuracy
    convergence_test.go:76: 
        	Error Trace:	convergence_test.go:76
        	            				convergence_test.go:56
        	Error:      	Not equal: 
        	            	expected: 4
        	            	actual  : 3
        	Test:       	TestConvergenceAndAccuracy
```

**Comment by tonyyang-svail** [comment]
I introduced `convergence_test` in https://github.com/sql-machine-learning/sqlflow/pull/1504 to make sure SQLFlow trained model can reach 100% accuracy during prediction on the training data, where the training data size is tiny.

It is possible, though unlikely, that the test fails. By increasing the training epochs, we can reduce the probability of test failing.

**Comment by wangkuiyi** [comment]
> I introduced `convergence_test` in #1504 to make sure SQLFlow trained model can reach 100% accuracy during prediction on the training data, where the training data size is tiny.
> 
> It is possible, though unlikely, that the test fails. By increasing the training epochs, we can reduce the probability of test failing.

@tonyyang-svail Thanks for the explanation. However, the failure is due to panic, but not accuracy.

**Comment by tonyyang-svail** [comment]
@wangkuiyi From the log message in this issue, the error is asserting failing, not panic.

Btw, the link https://travis-ci.com/github/sql-machine-learning/sqlflow/jobs/337411228#L1250-L1335 doesn't contain the error message you posted. Did you rerun the build?

**Comment by wangkuiyi** [comment]
Yes. It was about the accuracy but not panic. 

And, yes, I restarted the build.  And the convergence test passed.

**Comment by wangkuiyi** [comment]
@tonyyang-svail I got a suggestion on testing convergence from https://github.com/byzhang.  He suggested we use a **well-separated** dataset, which means that the positive/negative samples are clearly far away from each other.  This makes the prediction result almost definitely deterministic.

**Comment by wangkuiyi** [comment]
It was @tonyyang-svail who crafted the sanity_check dataset used in this test.  It is likely that we need to make it more well-separated.

**Comment by sneaxiy** [comment]
> I got a suggestion on testing convergence from https://github.com/byzhang. He suggested we use a well-separated dataset

@wangkuiyi The dataset in covergence_test.go is already well-separated.

https://github.com/sql-machine-learning/sqlflow/blob/f15f0ddc1fe271b0347b4ca9afedfedaa54722a4/pkg/sql/testdata/sanity_check.go#L20-L32

https://github.com/sql-machine-learning/sqlflow/blob/f15f0ddc1fe271b0347b4ca9afedfedaa54722a4/pkg/sql/testdata/sanity_check.go#L35-L48

I guess it is because the non-deterministic random seed when training.
